### PR TITLE
Display the related broken trunk (under BROKEN TRUNK) and similar failures (under FLAKY) on Dr.CI

### PR DESCRIPTION
**Pull request by huydhn**
As part of https://github.com/pytorch/test-infra/issues/5139, I'm updating `hasSimilarFailures` and rename `isBrokenTrunk` to `getTrunkFailure` to return the related similar and trunk failures.  These related failures can then be displayed on Dr.CI to provide additional insight on how it classifies the results.

This PR covers the first two items, the rest will be on the next PR I'm working on.

* [x] Provide additional information about broken trunk failures
* [x] Provide additional information about similar flaky failures
* [ ] Provide additional information about unstable failures
* [ ] Provide additional information about other flaky failures including jobs suppressed by labels and infra flaky.

### Testing

On https://github.com/pytorch/pytorch/pull/124920,

```
curl --request POST \
--url "http://localhost:3000/api/drci/drci?prNumber=124920" \
--header "Authorization: TOKEN" \
--data 'repo=pytorch'
```

### :test_tube: See artifacts and rendered test results at [hud.pytorch.org/pr/124920](https://hud.pytorch.org/pr/124920)
* :page_facing_up: Preview [Python docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/124920/index.html)
* :page_facing_up: Preview [C++ docs built from this PR](https://docs-preview.pytorch.org/pytorch/pytorch/124920/cppdocs/index.html)
* :question: Need help or want to give feedback on the CI? Visit the [bot commands wiki](https://github.com/pytorch/pytorch/wiki/Bot-commands) or our [office hours](https://github.com/pytorch/pytorch/wiki/Dev-Infra-Office-Hours)

Note: Links to docs will display an error until the docs builds have been completed.

## :white_check_mark: You can merge normally! (2 Unrelated Failures)
As of commit a6516ea6789e12a1a80a8c8cc7ce63698d443821 with merge base 59a1f1f308545e3ac1d81940a51f8dc0db3d82d4 (<sub><sub>image</sub></sub>):
<details ><summary><b>FLAKY</b> - The following job failed but was likely due to flakiness present on trunk:</summary><p>

* [pull / linux-focal-py3_8-clang9-xla / test (xla, 1, 1, linux.12xlarge)](https://hud.pytorch.org/pr/pytorch/pytorch/124920#24281813823) ([gh](https://github.com/pytorch/pytorch/actions/runs/8842036653/job/24281813823))([similar failure](https://hud.pytorch.org/pytorch/pytorch/commit/333f095d0779ecf0ce489ceecff35404abde8581#24279656996))
    `test_all_cpu_tensor`
</p></details>
<details ><summary><b>BROKEN TRUNK</b> - The following job failed but was present on the merge base:</summary><p>👉 <b>Rebase onto the `viable/strict` branch to avoid these failures</b></p><p>

* [pull / linux-focal-cuda12.1-py3.10-gcc9-sm86 / test (default, 4, 5, linux.g5.4xlarge.nvidia.gpu)](https://hud.pytorch.org/pr/pytorch/pytorch/124920#24280934973) ([gh](https://github.com/pytorch/pytorch/actions/runs/8842036653/job/24280934973)) ([trunk failure](https://hud.pytorch.org/pytorch/pytorch/commit/59a1f1f308545e3ac1d81940a51f8dc0db3d82d4#24248383727))
    `inductor/test_cudagraph_trees.py::CudaGraphTreeTests::test_mutation_cudagraph_managed_tensors_config_backend_cudagraphs`
</p></details>

This comment was automatically generated by Dr. CI and updates every 15 minutes.

### CODE REVIEW

File: torchci/pages/api/drci/drci.ts [hunk 1]
```diff
@@ -428,11 +442,28 @@ function constructResultsJobsSections(
...
+
+    const relatedJob = relatedJobs.get(job.id);
+    // Show the related trunk failure for broken trunk or the similar failure for flaky
+    if (relatedJob !== undefined) {
+      const hudCommitUrl = `${hudBaseUrl}/${owner}/${repo}/commit/${relatedJob.head_sha}`;
+      const relatedJobLink = `${hudCommitUrl}#${relatedJob.id}`;
+      if (header === "BROKEN TRUNK") {
+        output += ` ([trunk failure](${relatedJobLink}))`;
+      } else if (header === "FLAKY") {
+        output += `([similar failure](${relatedJobLink}))`;
```
Comment (clee2000) [inline torchci/pages/api/drci/drci.ts:459]:
```suggestion
        output += ` ([similar failure](${relatedJobLink}))`;
```

File: torchci/pages/api/drci/drci.ts [hunk 2]
```diff
@@ -428,11 +442,28 @@ function constructResultsJobsSections(
...
+      } else {
+        output += `([related job](${relatedJobLink}))`;
```
Comment (clee2000) [inline torchci/pages/api/drci/drci.ts:461]:
```suggestion
        output += ` ([related job](${relatedJobLink}))`;
```

File: torchci/pages/api/drci/drci.ts [hunk 3]
```diff
@@ -428,11 +442,28 @@ function constructResultsJobsSections(
...
+      }
+    }
+
     if (job.failure_captures) {
-      output += `    \`${job.failure_captures[0]}\`\n`;
+      output += `\n    \`${job.failure_captures[0]}\`\n`;
```
Comment (clee2000) [inline torchci/pages/api/drci/drci.ts:466]: sanity check, i think you might need a `\n` to divide lines within a category if the failures captures doesnt exist

**Review by clee2000 — APPROVED** [review]
A couple of minor things but other wise looks fine

File: torchci/pages/api/drci/drci.ts (hunk 3 above)
Comment (huydhn) [reply torchci/pages/api/drci/drci.ts:466]: Oh, yeah, good catch.  it would screw up the format otherwise

**Pull request merged**